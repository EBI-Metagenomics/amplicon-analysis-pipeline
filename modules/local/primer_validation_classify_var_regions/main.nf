
process PRIMER_VALIDATION_CLASSIFY_VAR_REGIONS {
    tag "$meta.id"
    label 'very_light'
    // container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
    //     "https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:${params.mpt_version}":
    //     "biocontainers/mgnify-pipelines-toolkit:${params.mpt_version}" }"
    container "oras://community.wave.seqera.io/library/pip_mgnify-pipelines-toolkit:64432eed2c673687"

    input:
    tuple val(meta), path(cmsearch_deoverlap_out), path(concat_primers_fasta)

    output:
    tuple val(meta), path("*primer_validation.tsv"), emit: primer_validation_out
    tuple val(meta), path("*primers.fasta")        , emit: validated_primers
    path "versions.yml"                            , emit: versions

    script:
    """
    // TODO remove absolute path obviously, this is temporary
    python /hps/software/users/rdf/metagenomics/service-team/users/chrisata/mgnify-pipelines-toolkit/mgnify_pipelines_toolkit/analysis/amplicon/primer_val_classification.py -i $cmsearch_deoverlap_out -f $concat_primers_fasta -s ${meta.id}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: ${params.mpt_version}
    END_VERSIONS
    """
}
