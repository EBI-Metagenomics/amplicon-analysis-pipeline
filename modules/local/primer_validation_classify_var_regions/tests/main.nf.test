nextflow_process {

    name "Test Process PRIMER_VALIDATION_CLASSIFY_VAR_REGIONS"
    script "../main.nf"
    process "PRIMER_VALIDATION_CLASSIFY_VAR_REGIONS"
    tag "modules"
    tag "primer_validation_classify_var_regions"

    test("primer_validation_classify_var_regions - should have primers in F and R") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'ERR4334351', single_end:false ], // meta map
                    file('${projectDir}/modules/local/primer_validation_classify_var_regions/tests/data/ERR4334351.tblout.deoverlapped', checkIfExists: true),
                    file('${projectDir}/modules/local/primer_validation_classify_var_regions/tests/data/ERR4334351_permuted_primers.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.primer_validation_out).match("primer_validation_out") }
            )
        }

    }

}
