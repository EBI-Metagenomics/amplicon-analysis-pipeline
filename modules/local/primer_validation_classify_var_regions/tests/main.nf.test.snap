{
    "primer_validation_out": {
        "content": [
            [
                [
                    {
                        "id": "ERR4334351",
                        "single_end": false
                    },
                    "ERR4334351_primer_validation.tsv:md5,438b00766b1f24f02de89c99358f8972"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-09T15:16:17.626557"
    }
}